File: Development/WoodLotFrontEnd/src/app/entita/pagamento/pagamento.ts
import {Contadino} from "../contadino/contadino";

export class Pagamento {

  public id: number
  public importo: number
  public dataPagamento: Date
  public motivazioni: String
  public contadino: Contadino

  constructor(id: number, importo: number, dataPagamento: Date, motivazioni: String, contadino: Contadino) {
    this.id = id
    this.importo = importo
    this.dataPagamento = dataPagamento
    this.motivazioni = motivazioni
    this.contadino = contadino
  }
}

File: Development/WoodLotBackEnd/src/main/java/it/unisa/WoodLot/model/entity/Pagamento.java
package it.unisa.WoodLot.model.entity;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

/**
 * Classe che modella un pagamento a un contadino
 *
 * @author Alessia Ture
 */
@Data
@Entity
@Table(name = "pagamento")
public class Pagamento implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "importo")
    private double importo;

    @Column(name = "data_pagamento")
    private Date dataPagamento;

    @Column(name = "motivazioni")
    private String motivazioni;

    @OneToOne
    @JoinColumn()
    private Contadino contadino;
}
